src/components/listing-detail.js

```diff
@@ -121,6 +147,46 @@ class ListingsDetail extends Component {
     }
   }
 
+  handleSkipOnboarding(e) {
+    e.preventDefault()
+
+    this.handleMakeOffer(true)
+  }
+
+  async loadBuyerPurchases() {
+    try {
+      const { web3Account } = this.props
+      const listingIds = await origin.marketplace.getListings({
+        idsOnly: true,
+        purchasesFor: web3Account
+      })
+      const listingPromises = listingIds.map(listingId => {
+        return new Promise(async resolve => {
+          const listing = await getListing(listingId, true)
+          resolve({ listingId, listing })
+        })
+      })
+      const withListings = await Promise.all(listingPromises)
+      const offerPromises = await withListings.map(obj => {
+        return new Promise(async resolve => {
+          const offers = await origin.marketplace.getOffers(obj.listingId, {
+            for: web3Account
+          })
+          resolve(Object.assign(obj, { offers }))
+        })
+      })
+      const withOffers = await Promise.all(offerPromises)
+      const offersByListing = withOffers.map(obj => {
+        return obj.offers.map(offer => Object.assign({}, obj, { offer }))
+      })
+      const offersFlattened = [].concat(...offersByListing)
+
+      this.setState({ purchases: offersFlattened })
+    } catch (error) {
+      console.error(error)
+    }
+  }
```

Comment:
Truth 🍝  Very soon, we aren't going to need this code because we will be fetching this data from the indexing server and not need all this spaghetti. I would also like to just load everything a user cares about into Redux at the beginning anyway.